crates/bevy_render/src/view/visibility/mod.rs

```diff
@@ -269,12 +271,52 @@ pub fn calculate_bounds(
     mut commands: Commands,
     meshes: Res<Assets<Mesh>>,
     without_aabb: Query<(Entity, &Handle<Mesh>), (Without<Aabb>, Without<NoFrustumCulling>)>,
+    without_handle: Query<Entity, (With<Aabb>, Without<Handle<Mesh>>, Without<NoFrustumCulling>)>,
 ) {
     for (entity, mesh_handle) in &without_aabb {
-        if let Some(mesh) = meshes.get(mesh_handle) {
-            if let Some(aabb) = mesh.compute_aabb() {
-                commands.entity(entity).insert(aabb);
+        let Some(mesh) = meshes.get(mesh_handle) else { continue };
+        let Some(aabb) = mesh.compute_aabb() else { continue };
+        commands.entity(entity).insert(aabb);
+    }
+    for entity in &without_handle {
+        commands.entity(entity).remove::<Aabb>();
+    }
+}
+
+pub fn update_bounds(
+    mut commands: Commands,
+    meshes: Res<Assets<Mesh>>,
+    mut events: EventReader<AssetEvent<Mesh>>,
+    mut with_aabb: Query<(Entity, &Handle<Mesh>, &mut Aabb), Without<NoFrustumCulling>>,
+    mut changed: Local<HashMap<HandleId, Aabb>>,
+    mut removed: Local<HashSet<HandleId>>,
+) {
+    changed.clear();
+    removed.clear();
+
+    for mesh_event in &mut events {
+        match mesh_event {
+            AssetEvent::Created { handle } | AssetEvent::Modified { handle } => {
+                let Some(mesh) = meshes.get(handle) else { continue };
+                let Some(aabb) = mesh.compute_aabb() else { continue };
+                changed.insert(handle.id(), aabb);
             }
+            AssetEvent::Removed { handle } => {
+                removed.insert(handle.id());
+            }
+        };
+    }
+
+    if changed.is_empty() && removed.is_empty() {
+        return;
+    }
+
+    // FIXME: Include entities that have changed their Handle<Mesh> component
+    for (entity, mesh_handle, mut entity_aabb) in &mut with_aabb {
+        if let Some(aabb) = changed.get(&mesh_handle.id()) {
+            *entity_aabb = *aabb;
+        } else if removed.contains(&mesh_handle.id()) {
+            commands.entity(entity).remove::<Aabb>();
```

Comment:
is it possible to get both a Created and Removed asset event for the same handle?

Reply:
I'm not certain, but I think we should probably code assuming this is possible, even if the current implementation happens to not make this possible (due to system ordering / deferred ops / etc).